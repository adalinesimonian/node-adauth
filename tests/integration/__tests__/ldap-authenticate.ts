import ADAuth from '../../../lib/adauth'

let auth: ADAuth

beforeEach(async () => {
  auth = await ADAuth.create({
    url: 'ldap://ldap.forumsys.com:389',
    domainDN: 'dc=example,dc=com',
    searchBase: 'dc=example,dc=com',
    // Test server is not an AD server, so search filter is manually specified.
    searchFilterBySAN: '(uid=riemann)',
    cache: true,
    groupSearchFilter: '(member={{dn}})',
    groupSearchBase: 'dc=example,dc=com',
  })

  auth.on('error', error => {
    console.warn(error)
    // TODO: auth.close() doesn't do anything here
  })
})

afterEach(async () => {
  await auth.dispose()
})

test('Can authenticate against test LDAP server', async () => {
  // Test server is not an AD server, so DN is specified in place of username so
  // that binding will work properly.
  const user = await auth.authenticate(
    'uid=riemann,dc=example,dc=com',
    'password'
  )
  expect(user).toBeTruthy()
})

test('Can re-authenticate against test LDAP server', async () => {
  const user1 = await auth.authenticate(
    'uid=riemann,dc=example,dc=com',
    'password'
  )
  expect(user1).toBeTruthy()
  const user2 = await auth.authenticate(
    'uid=riemann,dc=example,dc=com',
    'password'
  )
  expect(user2).toBeTruthy()
  expect(user1).toEqual(user2)
})
