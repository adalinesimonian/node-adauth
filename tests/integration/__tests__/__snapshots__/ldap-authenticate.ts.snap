// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With invalid credentials Fails authentication 1`] = `[InvalidDnSyntaxError: invalid DN]`;

exports[`With valid credentials Can authenticate against test LDAP server 1`] = `
Object {
  "_raw": undefined,
  "cn": "Bernhard Riemann",
  "controls": Array [],
  "dn": "uid=riemann,dc=example,dc=com",
  "groups": Array [],
  "mail": "riemann@ldap.forumsys.com",
  "objectClass": Array [
    "inetOrgPerson",
    "organizationalPerson",
    "person",
    "top",
  ],
  "objectGUID": undefined,
  "objectSid": undefined,
  "sn": "Riemann",
  "uid": "riemann",
  "userPassword": "{sha}W6ph5Mm5Pz8GgiULbPgzG37mj9g=",
}
`;

exports[`With valid credentials Can re-authenticate against test LDAP server 1`] = `
Object {
  "_raw": undefined,
  "cn": "Bernhard Riemann",
  "controls": Array [],
  "dn": "uid=riemann,dc=example,dc=com",
  "groups": Array [],
  "mail": "riemann@ldap.forumsys.com",
  "objectClass": Array [
    "inetOrgPerson",
    "organizationalPerson",
    "person",
    "top",
  ],
  "objectGUID": undefined,
  "objectSid": undefined,
  "sn": "Riemann",
  "uid": "riemann",
  "userPassword": "{sha}W6ph5Mm5Pz8GgiULbPgzG37mj9g=",
}
`;
